src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -90,23 +95,23 @@ public Path which(String fn) {
     protected Process createProcess() throws IOException {
         String[] commands = getCommand();
         logger.atTrace().kv("decorated command", String.join(" ", commands)).log();
+        ProcessBuilderForWin32 winPb = new ProcessBuilderForWin32();
         if (needToSwitchUser()) {
-            return createRunasProcess(commands);
-        } else {
-            ProcessBuilder pb = new ProcessBuilder();
-            pb.environment().putAll(environment);
-            return pb.directory(dir).command(commands).start();
+            String username = userDecorator.getUser();
+            winPb.user(username, new String(getPassword(username)));
+            // When environment is constructed it inherits current process env
+            // Clear the env in this case because later we'll load the given user's env instead
+            winPb.environment().clear();
         }
-    }
-
-    private Process createRunasProcess(String... commands) throws IOException {
-        // Expect username in format: DOMAIN\UserName
-        String username = userDecorator.getUser();
-        ProcessBuilderForWin32 winPb = new ProcessBuilderForWin32();
-        winPb.environment().clear();
         winPb.environment().putAll(environment);
-        winPb.user(username, new String(getPassword(username)));
-        return winPb.directory(dir).command(commands).start();
+        Process process = winPb.directory(dir).command(commands).start();
+        // calling attachConsole right after a process is launched will fail with invalid handle error
+        // waiting a bit ensures that we get the process handle from the pid
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException ignored) {
+        }
```

Comment:
interrupt the thread; very important in this location. `Thread.currentThread().interrupt()`

Reply:
Added